public/JavaScript/gallery.js

```diff
@@ -0,0 +1,33 @@
+document.addEventListener('DOMContentLoaded', () => {
+  let selectedAnimalType = null
+
+  // Function to hide all gallery items
+  const hideAllAnimals = (elements) => {
+    elements.forEach(item => {
+      item.style.display = 'none'
+    })
+  }
+  // Function to display selected animal
+  const showSelectedAnimals = () => {
+    hideAllAnimals(document.querySelectorAll('.gallery-items'))
+    if (selectedAnimalType) {
+      document.querySelectorAll(`.gallery-items[data-animal-type="${selectedAnimalType}"]`).forEach(item => {
+        item.style.display = 'block'
+      })
+    }
+  }
+  // Target animal type
+  document.querySelectorAll('[data-animal-type]').forEach(filterOption => {
+    filterOption.addEventListener('click', () => {
+      selectedAnimalType = filterOption.dataset.animalType
```

Comment:
When I click on an item, I change an almost global variable. So an action here, changes the state of the app somewhere else.

These kind of side-effects should be avoided.